### PR TITLE
c-kermit: Enable authentication and encryption

### PR DESCRIPTION
**Pull request by johnsonjh**
This enables authentication and encryption using OpenSSL 1.1.

Tested and verified working secure operations using SSL and TLS on START-TLS FTP, IKSD, and KRB TELNET services using both Linux and macOS X Ventura.

---

Note that no patches will be necessary after the release of C-Kermit 10, which will build using OpenSSL 3.

I think it would be better to bump this formula to C-Kermit 10 now, since this current (and very) "stable" 9.0 release came out in 2011. However, C-Kermit 10 is still only available as an alpha release upstream (although a new release is actually expected soon), and I know that it isn't in-line with Homebrew policies to use "unstable" pre-releases.

---

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c-kermit.rb [hunk 1]
```diff
@@ -61,3 +68,1264 @@ def install
      x = (int) ((stdin->_IO_read_end) - (stdin->_IO_read_ptr));
      debug(F101,"cmdconchk _IO_file_flags","",x);
  #else  /* _IO_file_flags */
+
+diff --git a/ck_ssl.c b/ck_ssl.c
```
Comment (carlocab) [inline Formula/c-kermit.rb:72]: Where is this patch from?

Comment (johnsonjh) [reply Formula/c-kermit.rb:72]: Based on https://github.com/BAN-AI-Communications/c-kermit/commit/fd0d853d76227d7929cabb63cae6ba57b295f633.diff which overall is similar to upstreams changes.  (It's not trivially possible to directly diff against the upstream simply there have been extensive changes over the last 12 years. The upstream changelog is at https://kermitproject.org/ckupdates.html where the first 50% describes the differences from this release to the current state.)

**Comment by johnsonjh** [comment]
Should now build in CI successfully.

File: Formula/c-kermit.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/c-kermit.rb:72]: I think we should just wait for an upstream release instead of cherry picking commits for extra features.

**Comment by johnsonjh** [comment]
All CI now passing.  

@SMillerDev 

The patch seems large since I made it the way the documentation states, using git diff and then appending those edits to the formula file, which gives (IMO) **waaaaay** too much context.

The diff made with GNU diff -rU0 is only about 220 lines excluding the headers (or about 350 including them).  The patch made by `git diff` is almost 1500 lines, all useless context.

I can change the patch to this one if that's better?  

```diff
diff --color -rU0 ./ckcftp.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckcftp.c
--- ./ckcftp.c	2011-07-14 12:17:30.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckcftp.c	2023-05-15 04:14:05.079128901 -0400
@@ -10199 +10199 @@
-        ssl_ftp_ctx=SSL_CTX_new(SSLv3_client_method());
+        ssl_ftp_ctx=SSL_CTX_new(TLS_client_method());
@@ -10207 +10207 @@
-                                  SSLv3_client_method());
+                                  TLS_client_method());
diff --color -rU0 ./ckcpro.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckcpro.c
--- ./ckcpro.c	2009-09-25 12:17:15.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckcpro.c	2023-05-15 04:14:05.079128901 -0400
@@ -10 +10 @@
-"C-Kermit Protocol Module 8.0.160, 12 Aug 2007";
+"C-Kermit Protocol Module 9.0.160, 16 Oct 2009";
@@ -21 +21 @@
-  Copyright (C) 1985, 2007,
+  Copyright (C) 1985, 2009,
@@ -171,0 +172 @@
+  extern int bctf;
@@ -506,0 +508,2 @@
+
+    if (!bctf) {		     /* Block check 3 forced on all packets */
@@ -508,4 +511,4 @@
-    if (epktflg) {			/* If because of E-PACKET command */
-	b1 = bctl; b2 = bctu;		/* Save block check type */
-	bctl = bctu = 1;		/* set it to 1 */
-    }
+	if (epktflg) {			/* If because of E-PACKET command */
+	    b1 = bctl; b2 = bctu;	/* Save block check type */
+	    bctl = bctu = 1;		/* set it to 1 */
+	}
@@ -512,0 +516 @@
+    }
@@ -513,0 +518 @@
+    if (!bctf) {		     /* Block check 3 forced on all packets */
@@ -515,3 +520,4 @@
-    if (epktflg) {			/* Restore the block check */
-	epktflg = 0;
-	bctl = b1; bctu = b2;
+	if (epktflg) {			/* Restore the block check */
+	    epktflg = 0;
+	    bctl = b1; bctu = b2;
+	}
@@ -948 +954 @@
-	    p =zgtdir();
+	    p = zgtdir();
@@ -1132 +1138 @@
-	debug(F100,"C-Kermit BYE - Loggin out...","",0);
+	debug(F100,"C-Kermit BYE - Logging out...","",0);
@@ -1776,2 +1782,8 @@
-    bctu = bctr;			/* switch to agreed-upon block check */
-    bctl = (bctu == 4) ? 2 : bctu;	/* Set block-check length */
+    if (bctf) {
+	bctu = 3;
+	bctl = 3;
+    } else {
+	bctu = bctr;			/* switch to agreed-upon block check */
+	bctl = (bctu == 4) ? 2 : bctu;	/* Set block-check length */
+    }
+
@@ -2929,2 +2941,7 @@
-	bctu = bctr;			/* Switch to agreed-upon block check */
-	bctl = (bctu == 4) ? 2 : bctu;	/* Set block-check length */
+	if (bctf) {
+	    bctu = 3;
+	    bctl = 3;
+	} else {
+	    bctu = bctr;	       /* switch to agreed-upon block check */
+	    bctl = (bctu == 4) ? 2 : bctu; /* Set block-check length */
+	}
diff --color -rU0 ./ckctel.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckctel.c
--- ./ckctel.c	2010-03-16 14:32:40.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckctel.c	2023-05-15 04:14:05.088129033 -0400
@@ -4318,0 +4319,3 @@
+#ifdef MACOSX
+    void ck_tn_enc_stop();
+#endif
diff --color -rU0 ./ck_ssl.c /tmp/c-kermit-20230515-3982872-yxuxdy/ck_ssl.c
--- ./ck_ssl.c	2011-07-06 09:03:32.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ck_ssl.c	2023-05-15 04:14:05.078128886 -0400
@@ -304 +304 @@
-                       ctx->error);
+                       X509_STORE_CTX_get_error(ctx));
@@ -935,0 +936 @@
+    BIGNUM *p, *g;
@@ -939,3 +940,3 @@
-    dh->p=BN_bin2bn(dh512_p,sizeof(dh512_p),NULL);
-    dh->g=BN_bin2bn(dh512_g,sizeof(dh512_g),NULL);
-    if ((dh->p == NULL) || (dh->g == NULL))
+    p=BN_bin2bn(dh512_p,sizeof(dh512_p),NULL);
+    g=BN_bin2bn(dh512_g,sizeof(dh512_g),NULL);
+    if ((p == NULL) || (g == NULL))
@@ -942,0 +944 @@
+    DH_set0_pqg(dh, p, NULL, g);
@@ -949,0 +952 @@
+    BIGNUM *p, *g;
@@ -953,3 +956,3 @@
-    dh->p=BN_bin2bn(dh768_p,sizeof(dh768_p),NULL);
-    dh->g=BN_bin2bn(dh768_g,sizeof(dh768_g),NULL);
-    if ((dh->p == NULL) || (dh->g == NULL))
+    p=BN_bin2bn(dh768_p,sizeof(dh768_p),NULL);
+    g=BN_bin2bn(dh768_g,sizeof(dh768_g),NULL);
+    if ((p == NULL) || (g == NULL))
@@ -956,0 +960 @@
+    DH_set0_pqg(dh, p, NULL, g);
@@ -963,0 +968 @@
+    BIGNUM *p, *g;
@@ -967,3 +972,3 @@
-    dh->p=BN_bin2bn(dh1024_p,sizeof(dh1024_p),NULL);
-    dh->g=BN_bin2bn(dh1024_g,sizeof(dh1024_g),NULL);
-    if ((dh->p == NULL) || (dh->g == NULL))
+    p=BN_bin2bn(dh1024_p,sizeof(dh1024_p),NULL);
+    g=BN_bin2bn(dh1024_g,sizeof(dh1024_g),NULL);
+    if ((p == NULL) || (g == NULL))
@@ -970,0 +976 @@
+    DH_set0_pqg(dh, p, NULL, g);
@@ -977,0 +984 @@
+    BIGNUM *p, *g;
@@ -981,3 +988,3 @@
-    dh->p=BN_bin2bn(dh1536_p,sizeof(dh1536_p),NULL);
-    dh->g=BN_bin2bn(dh1536_g,sizeof(dh1536_g),NULL);
-    if ((dh->p == NULL) || (dh->g == NULL))
+    p=BN_bin2bn(dh1536_p,sizeof(dh1536_p),NULL);
+    g=BN_bin2bn(dh1536_g,sizeof(dh1536_g),NULL);
+    if ((p == NULL) || (g == NULL))
@@ -984,0 +992 @@
+    DH_set0_pqg(dh, p, NULL, g);
@@ -991,0 +1000 @@
+    BIGNUM *p, *g;
@@ -995,3 +1004,3 @@
-    dh->p=BN_bin2bn(dh2048_p,sizeof(dh2048_p),NULL);
-    dh->g=BN_bin2bn(dh2048_g,sizeof(dh2048_g),NULL);
-    if ((dh->p == NULL) || (dh->g == NULL))
+    p=BN_bin2bn(dh2048_p,sizeof(dh2048_p),NULL);
+    g=BN_bin2bn(dh2048_g,sizeof(dh2048_g),NULL);
+    if ((p == NULL) || (g == NULL))
@@ -998,0 +1008 @@
+    DH_set0_pqg(dh, p, NULL, g);
@@ -1057,2 +1067,6 @@
-    if (ssl->expand == NULL || ssl->expand->meth == NULL)
-        printf("Compression: None\r\n");
+#ifndef OPENSSL_NO_COMP
+    const COMP_METHOD *x = SSL_get_current_expansion(ssl);
+    if (!x)
+#endif /* ifndef OPENSSL_NO_COMP */
+	printf("Compression: None\r\n");
+#ifndef OPENSSL_NO_COMP
@@ -1060 +1074 @@
-        printf("Compression: %s\r\n",ssl->expand->meth->name);
+        printf("Compression: %s\r\n", SSL_COMP_get_name(x));
@@ -1061,0 +1076 @@
+#endif /* ifndef OPENSSL_NO_COMP */
@@ -1460 +1475 @@
-    if (cm != NULL && cm->type != NID_undef) {
+    if (cm != NULL && COMP_get_type(cm) != NID_undef) {
@@ -1463,0 +1479 @@
+#if 0 /* COMP_rle has apparently been removed in OpenSSL 1.1 */
@@ -1466,0 +1483 @@
+#endif
@@ -1486,6 +1503,3 @@
-        rc1 = RAND_egd(ssl_rnd_file);
-        debug(F111,"ssl_once_init","RAND_egd()",rc1);
-        if ( rc1 <= 0 ) {
-            rc2 = RAND_load_file(ssl_rnd_file, -1);
-            debug(F111,"ssl_once_init","RAND_load_file()",rc1);
-        }
+        rc1 = -1;
+        rc2 = RAND_load_file(ssl_rnd_file, -1);
+        debug(F111,"ssl_once_init","RAND_load_file()",rc1);
@@ -1581,0 +1596 @@
+#ifndef OPENSSL_NO_SSL3_METHOD
@@ -1582,0 +1598 @@
+#endif /* ifndef OPENSSL_NO_SSL3_METHOD */
@@ -1584,0 +1601 @@
+#ifndef OPENSSL_NO_SSL3_METHOD
@@ -1585,0 +1603 @@
+#endif /* ifndef OPENSSL_NO_SSL3_METHOD */
@@ -1590 +1608 @@
-            tls_ctx=(SSL_CTX *)SSL_CTX_new(TLSv1_client_method());
+            tls_ctx=(SSL_CTX *)SSL_CTX_new(TLS_client_method());
@@ -1592 +1610 @@
-            tls_ctx=(SSL_CTX *)SSL_CTX_new(SSLv23_client_method());
+            tls_ctx=(SSL_CTX *)SSL_CTX_new(TLS_client_method());
@@ -1595,2 +1613,2 @@
-                debug(F110,"ssl_tn_init","SSLv23_client_method failed",0);
-                tls_ctx=(SSL_CTX *)SSL_CTX_new(SSLv3_client_method());
+                debug(F110,"ssl_tn_init","TLS_client_method failed",0);
+                tls_ctx=(SSL_CTX *)SSL_CTX_new(TLSv1_client_method());
@@ -1614 +1632 @@
-                ssl_ctx=(SSL_CTX *)SSL_CTX_new(SSLv3_server_method());
+                ssl_ctx=(SSL_CTX *)SSL_CTX_new(SSLv23_server_method());
@@ -1617 +1635 @@
-                debug(F110,"ssl_tn_init","SSLv3_server_method failed",0);
+                debug(F110,"ssl_tn_init","SSLv23_server_method failed",0);
@@ -2164 +2182 @@
-            tls_http_ctx=(SSL_CTX *)SSL_CTX_new(SSLv3_client_method());
+            tls_http_ctx=(SSL_CTX *)SSL_CTX_new(SSLv23_client_method());
@@ -2578 +2596 @@
-    X509_OBJECT obj;
+    X509_OBJECT *obj;
@@ -2644 +2662,5 @@
-    memset((char *)&obj, 0, sizeof(obj));
+    obj = X509_OBJECT_new();
+    if (!obj) {
+        X509_STORE_CTX_free(store_ctx);
+        return ok;
+    }
@@ -2646 +2668 @@
-    rc = X509_STORE_get_by_subject(store_ctx, X509_LU_CRL, subject, &obj);
+    rc = X509_STORE_get_by_subject(store_ctx, X509_LU_CRL, subject, obj);
@@ -2648 +2670 @@
-    crl = obj.data.crl;
+    crl = X509_OBJECT_get0_X509_CRL(obj);
@@ -2656 +2678 @@
-            X509_OBJECT_free_contents(&obj);
+            X509_OBJECT_free(obj);
@@ -2669 +2691 @@
-            X509_OBJECT_free_contents(&obj);
+            X509_OBJECT_free(obj);
@@ -2678 +2700 @@
-            X509_OBJECT_free_contents(&obj);
+            X509_OBJECT_free(obj);
@@ -2682 +2703,0 @@
-        X509_OBJECT_free_contents(&obj);
@@ -2689 +2710,6 @@
-    memset((char *)&obj, 0, sizeof(obj));
+    X509_OBJECT_free(obj);
+    obj = X509_OBJECT_new();
+    if (!obj) {
+        X509_STORE_CTX_free(store_ctx);
+        return ok;
+    }
@@ -2691 +2717 @@
-    rc = X509_STORE_get_by_subject(store_ctx, X509_LU_CRL, issuer, &obj);
+    rc = X509_STORE_get_by_subject(store_ctx, X509_LU_CRL, issuer, obj);
@@ -2693 +2719 @@
-    crl = obj.data.crl;
+    crl = X509_OBJECT_get0_X509_CRL(obj);
@@ -2701 +2727 @@
-            if (ASN1_INTEGER_cmp(revoked->serialNumber,
+            if (ASN1_INTEGER_cmp(X509_REVOKED_get0_serialNumber(revoked),
@@ -2704 +2730 @@
-                serial = ASN1_INTEGER_get(revoked->serialNumber);
+                serial = ASN1_INTEGER_get(X509_REVOKED_get0_serialNumber(revoked));
@@ -2709 +2735 @@
-                X509_OBJECT_free_contents(&obj);
+                X509_OBJECT_free(obj);
@@ -2713 +2738,0 @@
-        X509_OBJECT_free_contents(&obj);
@@ -2714,0 +2740 @@
+    X509_OBJECT_free(obj);
@@ -4332,0 +4359 @@
+    const ASN1_BIT_STRING *peer_sig, *file_sig;
@@ -4335,0 +4363 @@
+    X509_get0_signature(&peer_sig, NULL, peer_cert);
@@ -4346 +4374,2 @@
-        if (!ASN1_STRING_cmp(peer_cert->signature, file_cert->signature))
+        X509_get0_signature(&file_sig, NULL, file_cert);
+        if (!ASN1_STRING_cmp(peer_sig, file_sig))
diff --color -rU0 ./ckuath.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckuath.c
--- ./ckuath.c	2011-06-13 13:26:54.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckuath.c	2023-05-15 04:14:05.080128916 -0400
@@ -4526 +4526 @@
-            com_err("k5_get_ccache",r,"while getting default ccache");
+            /* com_err("k5_get_ccache",r,"while getting default ccache"); */
@@ -10930 +10930 @@
-            com_err("ck_krb5_get_cc_name",code,"while getting default ccache");
+            /* com_err("ck_krb5_get_cc_name",code,"while getting default ccache"); */
diff --color -rU0 ./ckucmd.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckucmd.c
--- ./ckucmd.c	2011-07-14 08:14:37.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckucmd.c	2023-05-15 04:14:05.078128886 -0400
@@ -7373 +7373 @@
-#ifdef _IO_file_flags			/* Linux */
+#ifdef _IO_EOF_SEEN			/* Linux */
Only in /tmp/c-kermit-20230515-3982872-yxuxdy: ckucmd.c.orig
diff --color -rU0 ./ckucns.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckucns.c
--- ./ckucns.c	2010-03-01 15:02:16.000000000 -0500
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckucns.c	2023-05-15 04:14:05.083128960 -0400
@@ -197,0 +198 @@
+#include "ckuath.h"
diff --color -rU0 ./ckufio.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckufio.c
--- ./ckufio.c	2011-08-20 18:49:50.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckufio.c	2023-05-15 04:14:05.084128975 -0400
@@ -43,0 +44 @@
+#include "ckuath.h"
diff --color -rU0 ./ckupty.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckupty.c
--- ./ckupty.c	2011-06-13 11:34:13.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckupty.c	2023-05-15 04:14:05.084128975 -0400
@@ -69,0 +70,4 @@
+#ifdef MACOSX
+#include <util.h>
+#endif
+
diff --color -rU0 ./ckutio.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckutio.c
--- ./ckutio.c	2011-08-20 17:22:35.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckutio.c	2023-05-15 04:14:05.086129004 -0400
@@ -374,0 +375,4 @@
+#ifdef MACOSX
+#include <util.h>
+#endif /* MACOSX */
+
@@ -14663,0 +14668,3 @@
+#ifdef MACOSX
+    int openpty(int *aprimary, int *areplica, char *name, struct termios *termp, struct winsize *winp);
+#endif
diff --color -rU0 ./ckuus3.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckuus3.c
--- ./ckuus3.c	2011-06-26 14:20:07.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckuus3.c	2023-05-15 04:14:05.081128930 -0400
@@ -10308,0 +10309 @@
+		  void ck_tn_enc_start();
@@ -10319,0 +10321 @@
+		  void ck_tn_enc_stop();
diff --color -rU0 ./ckuusr.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckuusr.c
--- ./ckuusr.c	2011-06-09 14:35:17.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckuusr.c	2023-05-15 04:14:05.082128945 -0400
@@ -95,0 +96 @@
+#include "ckuath.h"
diff --color -rU0 ./ckuusx.c /tmp/c-kermit-20230515-3982872-yxuxdy/ckuusx.c
--- ./ckuusx.c	2011-06-17 10:11:34.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/ckuusx.c	2023-05-15 04:14:05.083128960 -0400
@@ -46,0 +47,3 @@
+#ifdef CK_NCURSES
+#include <ncurses.h>
+#endif /* ifdef CK_NCURSES */
@@ -65,0 +69,3 @@
+#ifdef CK_NCURSES
+#include <term.h>
+#endif /* ifdef CK_NCURSES */
@@ -66,0 +73 @@
+
Only in /tmp/c-kermit-20230515-3982872-yxuxdy: .git
diff --color -rU0 ./makefile /tmp/c-kermit-20230515-3982872-yxuxdy/makefile
--- ./makefile	2011-08-21 11:12:07.000000000 -0400
+++ /tmp/c-kermit-20230515-3982872-yxuxdy/makefile	2023-05-15 04:14:05.089129048 -0400
@@ -2137 +2137 @@
-	-DCKCPU=\"\\\"$${MACCPU}\\\"\" \
+	-Wno-implicit-function-declaration -DCKCPU=\"\\\"$${MACCPU}\\\"\" \
@@ -2139 +2139 @@
-	"LIBS= $$HAVE_KRB5CONFIG -lssl -lcrypto -lpam -lncurses $(LIBS)"
+	"LIBS= $$HAVE_KRB5CONFIG -lssl -lcrypto -lpam -lncurses -lresolv $(LIBS)"
@@ -6305 +6305 @@
-	K5CRYPTO=''; \
+	K5CRYPTO='-lk5crypto'; \
@@ -6313 +6313 @@
-	COM_ERR=''; \
+	COM_ERR='-lcom_err'; \
@@ -6317 +6317 @@
-	GSSAPILIB='-lgssapi'; \
+	GSSAPILIB='-lgssapi_krb5'; \
@@ -6332 +6332 @@
-	-lcrypto $$GSSAPILIB -lkrb5 $$K5CRYPTO $$COM_ERR $(LIBS)" ; \
+	-lcrypto $$GSSAPILIB -lkrb5 $$K5CRYPTO $$COM_ERR -lncurses $(LIBS)" ; \
```

File: Formula/c-kermit.rb (hunk 1 above)
Comment (johnsonjh) [reply Formula/c-kermit.rb:72]: See https://github.com/Homebrew/homebrew-core/pull/130987#issuecomment-1547405456 for what I think is a better patch to append.

**Comment by johnsonjh** [comment]
There aren't too many KRBized TELNET servers publicly available, nor IKS, although I know it's working for IKSD, and you can also verify the features work via any publicly accessible START-TLS-only FTP server, if need be, for example:

```
> SET FTP VERBOSE-MODE ON
> SET FTP DATA-PROTECTION-LEVEL CONFIDENTIAL
> SET FTP AUTHTYPE TLS
> FTP public.dhe.ibm.com
Connected to public.dhe.ibm.com.
ProFTPD Server (proftpd) [170.225.126.18]
AUTH TLS successful
TLS accepted as authentication type
[TLS - ECDHE-RSA-AES128-GCM-SHA256 TLSv1.2 Kx=ECDH     Au=RSA  Enc=AESGCM(128) Mac=AEAD
Compression: None
FTP Command channel is Private (encrypted)
Protection set to Private
FTP Data channel is Private (encrypted)
 Name (public.dhe.ibm.com:jhj): anonymous
Anonymous login ok, send your complete email address as your password
 Password: 
Anonymous access granted, restrictions apply
Switching LOCUS for file-management commands to REMOTE (HELP LOCUS for info).
Remote system type is UNIX.
Default transfer mode is BINARY
> 
```

**Comment by johnsonjh** [comment]
Actually, looks like I can clean up a bit more, due to some change that was just whitespace...

**Comment by johnsonjh** [comment]
Yeah, I'll clean it up a bit more, and then I'll re-open.  Sorry for the false alarm.

**Pull request closed without merging**